### PR TITLE
Remove old docs

### PR DESCRIPTION
**Pull request by gj**
PR checklist:

- [ ] Update Go quickstart to conform to others & add it to test-quickstarts target.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
+	path = docs/examples/roles/sqlalchemy/basic
+	url = https://github.com/osohq/oso-sqlalchemy-roles-guide.git
+	branch = basic_roles_complete
+[submodule "docs/examples/quickstart/go"]
+	path = docs/examples/quickstart/go
+	url = https://github.com/osohq/oso-go-quickstart.git
+[submodule "docs/examples/quickstart/java"]
+	path = docs/examples/quickstart/java
 	url = https://github.com/osohq/oso-java-quickstart.git
-        branch = main
```
Comment (samscott89) [inline .gitmodules:4]: This doesn't work for me without specifying all the branches as main.

Comment (gj) [reply .gitmodules:4]: ah right; I think that's due to a difference in Git versions

Comment (gj) [reply .gitmodules:4]: fixed in b83a9c9c

Comment (samscott89) [reply .gitmodules:4]: Thank you kindly

**Review by samscott89 — APPROVED** [review]
I reviewed this as best as I could:

- ran the docs locally
- ran the docs tests
- looked at whatever changes seemed substantive. though there's a lot of noise with old docs files being "updated" with the move.

This looks good to me.

**Comment by gj** [comment]
> Edit to add: I don't understand what
> 
> > ```
> > Update Go quickstart to conform to others & add it to test-quickstarts target.
> > ```
> 
> is referring to on the PR checklist. This is not introduced by this PR, right?

Correct, just a general observation that we weren't including the Go quickstart in the `test-quickstarts` make target that gates releases.

**Pull request merged**